corehq/motech/generic_inbound/views.py

```diff
@@ -64,3 +79,52 @@ def get_create_item_data(self, create_form):
             "itemData": self._item_data(new_expression),
             "template": "base-api-config-template",
         }
+
+
+@method_decorator(toggles.GENERIC_INBOUND_API.required_decorator(), name='dispatch')
+class ConfigurableAPIEditView(BaseProjectSettingsView):
+    page_title = _("Edit API Configuration")
+    urlname = "configurable_api_edit"
+    template_name = "generic_inbound/api_edit.html"
+
+    @property
+    def api_id(self):
+        return self.kwargs['api_id']
+
+    @property
+    @memoized
+    def api(self):
+        try:
+            return ConfigurableAPI.objects.get(id=self.api_id)
+        except ConfigurableAPI.DoesNotExist:
+            raise Http404
+
+    @property
+    def page_url(self):
+        return reverse(self.urlname, args=(self.domain, self.api_id,))
+
+    def get_form(self):
+        if self.request.method == 'POST':
+            return ConfigurableAPIUpdateForm(self.request, self.request.POST, instance=self.api)
+        return ConfigurableAPIUpdateForm(self.request, instance=self.api)
+
+    @property
+    def main_context(self):
+        main_context = super(ConfigurableAPIEditView, self).main_context
+
+        main_context.update({
+            "form": self.get_form(),
+            "api_model": self.api,
+        })
+        return main_context
+
+    def post(self, request, domain, **kwargs):
+        form = self.get_form()
+        if form.is_valid():
+            form.save()
+            messages.success(request, _("API Configuration updated successfully."))
```

Comment:
This should probably redirect on success rather than `return get` so refreshing doesn't resubmit the form

Reply:
625d301999b